xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -463,3 +475,126 @@ func circuitBreakersFromCluster(cluster *v3clusterpb.Cluster) *uint32 {
 	}
 	return nil
 }
+
+// outlierConfigFromCluster extracts the relevant outlier detection
+// configuration from the received cluster resource. Returns nil if no
+// OutlierDetection field set in the cluster resource.
+func outlierConfigFromCluster(cluster *v3clusterpb.Cluster) (*OutlierDetection, error) {
+	od := cluster.GetOutlierDetection()
+	if od == nil {
+		return nil, nil
+	}
+	const (
+		defaultInterval                       = 10 * time.Second
+		defaultBaseEjectionTime               = 30 * time.Second
+		defaultMaxEjectionTime                = 300 * time.Second
+		defaultMaxEjectionPercent             = 10
+		defaultSuccessRateStdevFactor         = 1900
+		defaultEnforcingSuccessRate           = 100
+		defaultSuccessRateMinimumHosts        = 5
+		defaultSuccessRateRequestVolume       = 100
+		defaultFailurePercentageThreshold     = 85
+		defaultEnforcingFailurePercentage     = 0
+		defaultFailurePercentageMinimumHosts  = 5
+		defaultFailurePercentageRequestVolume = 50
+	)
+	// "The google.protobuf.Duration fields interval, base_ejection_time, and
+	// max_ejection_time must obey the restrictions in the
+	// google.protobuf.Duration documentation and they must have non-negative
+	// values." - A50
+	// TODO: I can find non negative examples for validation, but not "must obey the restrictions in the documentation"
+	var interval = defaultInterval
+	if i := od.GetInterval(); i != nil {
+		if err := i.CheckValid(); err != nil {
+			return nil, fmt.Errorf("outlier_detection.interval is invalid with error %v", err)
+		}
+		if interval = i.AsDuration(); interval < 0 {
```

Comment:
Nope. I promise I thought of this too :). Here's the checks in case you're curious: case x == nil:
		return invalidNil
	case secs < -absDuration:
		return invalidUnderflow
	case secs > +absDuration:
		return invalidOverflow
	case nanos <= -1e9 || nanos >= +1e9:
		return invalidNanosRange
	case (secs > 0 && nanos < 0) || (secs < 0 && nanos > 0):
		return invalidNanosSign